File: backend/api/routes/voucher.js
const express = require('express');
const router = express.Router();
const { 
  getVouchers, 
  getVoucherById, 
  createVoucher, 
  updateVoucher, 
  deleteVoucher 
} = require('../controllers/voucherController');
const { uploadVoucher } = require('../middleware/uploadMiddleware');
const { authMiddleware, roleMiddleware } = require('../middleware');

// Public routes
router.get('/', getVouchers);
router.get('/:id', getVoucherById);

// Admin routes
router.post('/', authMiddleware, roleMiddleware('admin'), uploadVoucher, createVoucher);
router.put('/:id', authMiddleware, roleMiddleware('admin'), uploadVoucher, updateVoucher);
router.patch('/:id', authMiddleware, roleMiddleware('admin'), uploadVoucher, updateVoucher);
router.delete('/:id', authMiddleware, roleMiddleware('admin'), deleteVoucher);

module.exports = router;

File: fastapi/src/routes/voucher.py
from fastapi import APIRouter
from src.controllers.voucher_controller import get_vouchers, get_voucher_by_id

router = APIRouter()


@router.get("/", response_model=dict)
async def get_vouchers_route():
    return await get_vouchers()


@router.get("/{voucher_id}", response_model=dict)
async def get_voucher_route(voucher_id: str):
    return await get_voucher_by_id(voucher_id)